components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -347,6 +349,14 @@ fn pow(lhs: &Option<Real>, rhs: &Option<Real>) -> Result<Option<Real>> {
     }
 }
 
+#[inline]
+#[rpn_fn]
+fn rand() -> Result<Option<Real>> {
```

Comment:
Use `metadata` to cache the rng. You can find an example here: https://github.com/tikv/tikv/blob/01decbeb5cdae45ee89bf352695e78eb81ba963a/components/tidb_query/src/rpn_expr/impl_compare_in.rs#L130
where you can store the Rng directly as metadata

Reply:
The metadata seem not support &mut. The rand gen fn  need borrow rng as mutable.
I checked the rpn doc and other metadata cache example, still have no idea.
 Could you please give me some suggestions?

Reply:
cached.

Reply:
Well done!